File: src/main/java/com/lsd/report/HtmlIndexWriter.java
package com.lsd.report;

import com.lsd.CapturedReport;
import com.lsd.properties.LsdProperties;
import lombok.SneakyThrows;
import lombok.extern.slf4j.Slf4j;

import java.io.File;
import java.nio.file.Files;
import java.nio.file.Path;
import java.util.List;

import static com.lsd.properties.LsdProperties.OUTPUT_DIR;

@Slf4j
public class HtmlIndexWriter {

    @SneakyThrows //TODO handle gracefully instead of propagating?
    public static Path writeToFile(List<CapturedReport> reports) {
        var outputDir = new File(LsdProperties.get(OUTPUT_DIR));
        outputDir.mkdir();
        Path path = new File(outputDir, "lsd-index.html").toPath();
        Files.write(path, new HtmlIndexRenderer().render(reports).getBytes());
        System.out.println("LSD Index:\nfile://" + path.toAbsolutePath());
        return path;
    }
}

File: src/main/kotlin/com/lsd/report/HtmlIndexWriter.kt
package com.lsd.report

import com.lsd.CapturedReport
import com.lsd.properties.LsdProperties
import com.lsd.properties.LsdProperties.OUTPUT_DIR
import java.io.File
import java.nio.file.Files
import java.nio.file.Path

object HtmlIndexWriter {

    fun writeToFile(reports: List<CapturedReport>): Path {
        File(LsdProperties[OUTPUT_DIR]).mkdir()
        return File(File(LsdProperties[OUTPUT_DIR]), "lsd-index.html").toPath().also {
            Files.write(it, HtmlIndexRenderer().render(reports).toByteArray())
            printLocationOfIndex(it)
        }
    }

    private fun printLocationOfIndex(path: Path) {
        println(
            """
        LSD Index:
        file://${path.toAbsolutePath()}
        """.trimIndent()
        )
    }
}